src/wurf/waf_resolve_context.py

```diff
@@ -124,6 +117,17 @@ def execute(self):
         for action in post_resolver_actions:
             action()
 
+    def post_recurse(self, node):
```

Comment:
Well, the previous solution worked by accident ;) But we need proper recursion since we call `ctx.is_toplevel()` when handling internal dependencies. Using `post_recurse` is like adding a last line in the resolve function of each wscript to call `dependency_manager.load_dependencies()`. This is the "waf way" to handle recursion.